tests/ui/redundant_field_names.rs

```diff
@@ -0,0 +1,28 @@
+#![warn(redundant_field_names)]
+#![allow(unused_variables)]
+
+mod foo {
+    pub const BAR: u8 = 0;
+}
+
+struct Person {
+    gender: u8,
+    age: u8,
+
+    buzz: u64,
+    foo: u8,
+}
+
+fn main() {
+    let gender: u8 = 42;
+    let age = 0;
+    let fizz: u64 = 0;
+
+    let me = Person {
+        gender: gender,
+        age: age,
+
+        buzz: fizz, //should be ok
+        foo: foo::BAR, //should be ok
```

Comment:
can you add a test for already using the shorthand?

Reply:
Yep, working on it.